src/task/TraitManager.ts

```diff
@@ -35,16 +37,29 @@ export class TraitManager {
 		}
 		return Promise.resolve(completions);
 	}
+	static async provideTraitProperties(traitName: string, position: vscode.Position): Promise<vscode.CompletionItem[]> {
+		const completions: vscode.CompletionItem[] = [];
+		const traitDefs = await TraitManager.retrieveTraitsDefinitions(traitName);
+		for (const property of traitDefs[0].properties) {
+			const propertyCompletion: vscode.CompletionItem = {
+				label: property.name,
+				sortText: SORT_PREFIX_TO_HAVE_COMPLETIONS_BEFORE_VARIABLES + property.name,
+				range: new vscode.Range(position, position)
+			};
+			completions.push(propertyCompletion);
+		}
+		return Promise.resolve(completions);
```

Comment:
I am confused by this line. An async function should not need to return something via Promise.resolve() but can simply return whatever variable it wants.

Reply:
seems both are working. I don't see clear advantages/drawbacks on the 2 possibilities. Changed to return the type directly instead fo the Promise.

Reply:
I think when in an async function it will do that automatically for you.